### DISCUSSION TITLE
Extending datatypes (ex xsd.DateTime)

### DISCUSSION THREAD

**Discussion by last-partizan** [mvantellingen/python-zeep#1436] [Ideas]
Hello, this is not an issue but rather question.

I'm using soap service which expect me to use date format like this `2016-08-14T00:00:00.000+03:00`

And does not accept format used by zeep. I found `zeep.xsd.DateTime` which converts values to strings, but not found a way to use it.

i subclassed zeep.xsd.DateTime, and tried this:

``` .python
c = zeep.Client(wsdl)
c.client.SomeMethod(SomeMethodRequest={'DateTime': MyDateTime(dt)}
```

and this is not working

so, how do i tell zeep to use my custom DateTime class?

**Comment by mvantellingen** [comment]
For now the easiest way is to update the zeep.xsd.builtins.default_types dictionary. You can override the builtin instance for the xsd:dateTime handling there. Bit hacky and not really future proof :-)

**Comment by last-partizan** [comment]
Thanks, that's better than my current solution using Plugin and parsing/replacing variable :)
